### PR TITLE
Make 'height' parameter strongly typed

### PR DESCRIPTION
**Pull request by Geod24**
By using a `struct` which `alias this` to the underlying (ulong) type,
we can guarantee that callers of the functions accepting `Height`
will need to provide a `Height` (or build one), while allowing
most manipulation of `Height` as a regular `ulong`.

### CODE REVIEW

**Review by TrustHenry — APPROVED** [review]
It's excluded the network API and checked its applied.

File: source/agora/common/Types.d [hunk 1]
```diff
@@ -58,3 +58,13 @@ public struct QuorumConfig
...
     public QuorumConfig[] quorums;
 }
+
+/// A type to ensure that height and other integer values aren't mixed
+public struct Height
+{
+    ///
+    public ulong value;
+
+    /// Provides implicit convertion to `ulong`
```
Comment (AndrejMitrovic) [inline source/agora/common/Types.d]:
```suggestion
    /// Provides implicit conversion to `ulong`
```

Comment (Geod24) [reply source/agora/common/Types.d]: Done

**Comment by AndrejMitrovic** [comment]
There's a bunch of `size_t block_height` I see in `BlockStorage.d` and `NetworkManager.d`

**Comment by linked0** [comment]
Could you explain the meaning of providing a `Height` instead of `ulong`?

**Comment by Geod24** [comment]
It avoids issues such as mistakenly passing a distance when a height is expected, for example.

**Comment by Geod24** [comment]
> There's a bunch of `size_t block_height` I see in `BlockStorage.d` and `NetworkManager.d`

Fixed.

**Comment by Geod24** [comment]
This PR seems to have uncovered a bug with the Serializer. I found it, and will address it soon. If you ACK the concept (and most of the change, bar the failures / conflict), I will fix and self-merge over the weekend.

**Comment by AndrejMitrovic** [comment]
ACK.

**Review by AndrejMitrovic — APPROVED** [review]

**Comment by Geod24** [comment]
Found the bug: https://github.com/bpfkorea/agora/pull/958

**Comment by Geod24** [comment]
Rebased on v0.x.x, will self merge when green.

**Pull request merged**